### PR TITLE
6 list comments

### PR DESCRIPTION
**Pull request by NeharikUmashankar**

### CODE REVIEW

File: src/components/Comments.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    getComments(review_id).then((result) => {
+      setComments(result.data.comments);
+      setCommentLoad(false);
+    });
+  }, []);
+
+  if (commentLoad) return <p>Comments loading, please wait....</p>;
+
+  if (Array.isArray(comments)) {
+    if (comments.length === 0) {
```
Comment (lewispricey) [inline src/components/Comments.jsx:19]: Great check to display a message if the theres no comments in the array

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      return <p>No comments</p>;
+    } else {
+      return (
+        <div className="comments">
+          {comments.map((comment) => {
+            return (
+              <div className="comment">
```
Comment (lewispricey) [inline src/components/Comments.jsx]: Is there a more semantic HTML tag you could use instead of a div here?

File: src/components/Comments.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { getComments } from "../api";
+
+const Comments = () => {
+  const { review_id } = useParams();
+  const [comments, setComments] = useState(undefined);
+  const [commentLoad, setCommentLoad] = useState(true);
```
Comment (lewispricey) [inline src/components/Comments.jsx:6]: Good use of the isLoading to indicate to the user if we're still waiting for content back

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work on this one so far Neharik!

Really like the length check to display no comments aswell as the isLoading as its so important on a slow connection a user knows if the page is supposed to be blank vs if things are still loading 👍 

While what you have works, It would be good to have the comments appear under the review to give these context. I like the idea of being able to toggle whether the comments show or not but have a think about how Youtube, LinkedIn ect show their comments on a individual post.

Please have a think about how you could render the comments under the article and raise another PR once done.

**Review by leekli — CHANGES_REQUESTED** [review]
Hey Neharik,

Well done so far! I can see all the functionality required is working here, and it's great to see you have a message of 'no comments' when the review has no comments yet.

However... I would really like to see the comments on the same page as the review I've clicked on. So on a single review page I should see: the review at the top, and the comments below. Please could you implement this. Once that's complete then you should find there's no need for the following route in your App.js:

```
<Route
          path="/reviews/:review_id/comments"
          element={<Comments />}
        ></Route>
```
as the single review page will take control of the comments.

It wouldn't be normal for a user to be able to go to a page such as /reviews/1/comments just to get the comments on their own.

Please look into this, implement, and submit an updated pull request review once complete.

**Review by leekli — APPROVED** [review]
Much better, thanks Neharik!

You are free to merge this branch.

**Pull request merged**